homeassistant/components/proxmoxve/__init__.py

```diff
@@ -219,14 +229,15 @@ def poll_api():
     )
 
 
-def parse_api_container_vm(status):
+def parse_api_container_vm(status: dict[str, str | int]) -> ProxmoxAPIData:
     """Get the container or vm api data and return it formatted in a dictionary.
 
     It is implemented in this way to allow for more data to be added for sensors
     in the future.
     """
-
-    return {"status": status["status"], "name": status["name"]}
+    return ProxmoxAPIData(
+        status=str(status["status"]), name=str(status["name"]), mem=int(status["mem"])
```

Comment:
If they are already correct values, there is no need to force convert them again.
If they are all strings, then only "mem" needs to be forced converted.

```suggestion
        status=status["status"], name=status["name"], mem=status["mem"]
```